graphene_django/relay/connection.py

```diff
@@ -139,45 +167,72 @@ def _handle_first_after(
     else:
         end_offset = start_offset + (first or 0)
 
-    trimmed_slice: SizedSliceable
-    has_previous_page: bool
-    has_next_page: bool
+    start: Optional[int]
+    stop: Optional[int]
 
     if end_offset is None:
-        trimmed_slice = sized_sliceable[start_offset:]
-        has_previous_page = start_offset > 0
-        has_next_page = False
+        start = start_offset
+        stop = None
+
+        def compute_slice(trimmed_slice):
+            has_previous_page: bool = start_offset > 0
+            has_next_page: bool = False
+
+            return trimmed_slice, has_previous_page, has_next_page
+
     else:
         # Slice off one more than we will be returning
-        intermediate_slice: SizedSliceable = sized_sliceable[
-            start_offset : end_offset + 1
-        ]
-        # Keep intermediate `intermediate_slice_length` variable to force QuerySet evaluation.
-        intermediate_slice_length: int = len(intermediate_slice)
+        start = start_offset
+        stop = end_offset + 1
 
-        trimmed_slice = intermediate_slice[: end_offset - start_offset]
-        trimmed_slice_length: int = len(trimmed_slice)
+        def compute_slice(intermediate_slice):
+            # Keep intermediate `intermediate_slice_length` variable to force QuerySet evaluation.
+            intermediate_slice_length: int = len(intermediate_slice)
 
-        has_next_page = intermediate_slice_length > trimmed_slice_length
+            trimmed_slice: SizedSliceable = intermediate_slice[
+                : end_offset - start_offset
+            ]
+            trimmed_slice_length: int = len(trimmed_slice)
 
-        # If the start offset is greater than zero, there is a previous page.
-        # However, if the provided `after` cursor is outside the bounds of the slice,
-        # enforce that `has_previous_page` is `True`.
-        has_previous_page = start_offset > 0
+            has_next_page: bool = intermediate_slice_length > trimmed_slice_length
 
-    edges = [
-        edge_type(
-            node=node,
-            cursor=offset_to_cursor(start_offset + index),
+            # If the start offset is greater than zero, there is a previous page.
+            # However, if the provided `after` cursor is outside the bounds of the slice,
+            # enforce that `has_previous_page` is `True`.
+            has_previous_page: bool = start_offset > 0
+
+            return trimmed_slice, has_previous_page, has_next_page
+
+    def compute_edges(result):
```

Comment:
Out of curiosity, why is this modification only necessary in the `_handle_first_after` case?

Reply:
In the `_handle_last_before` case, we need to calculate the `len` of the `sized_sliceable` to calculate `start` and `stop`, so we can't use dataloaders here unfortunately